src/gui/inputdevices/NewGtkInputDevice.cpp

```diff
@@ -323,12 +324,13 @@ bool NewGtkInputDevice::eventHandler(GdkEvent* event)
 	else if (event->type == GDK_BUTTON_PRESS || event->type == GDK_TOUCH_BEGIN)
 	{
 		input->copyAxes(event);
-		input->actionStart();
+		guint32 time = event->type == GDK_BUTTON_PRESS ? ((GdkEventButton *)event)->time : ((GdkEventTouch *)event)->time;  //or call gdk_event_get_time()
```

Comment:
Different events have different structures.. only the first few entries ( including event->type ) are the same and otherrwise you need to check the event->type and then cast to the right type and get the wanted field if present.See gdkEvents.h
Functions getEventTime and getEventState makes it easier.